.github/workflows/pull-request.yaml

```diff
@@ -33,6 +33,10 @@ jobs:
       with:
         commitlintRulesPath: ./commitlint.rules.js
         commitTitleMatch: 'true'
+
   build:
     needs: pr
     uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: write
```

Comment:
Why is `write` required here?

Reply:
> 

Because `build.yaml` needs to generate the artifacts and bump the version if it's running on `main`.

Reply:
Oh right, `build.yaml` pushes to the repo 🤦🏻‍♂️

https://github.com/jenstroeger/python-package-template/blob/1cd7aa0cb52b2395aeef33c8f46b914ac45b92e4/.github/workflows/build.yaml#L101-L105

although the [upload-artifact](https://github.com/actions/upload-artifact) Action doesn’t seem to need it?

Reply:
> although the [upload-artifact](https://github.com/actions/upload-artifact) Action doesn’t seem to need it?

No it doesn't.